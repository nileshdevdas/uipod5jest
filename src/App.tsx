import React from 'react';
import './App.css';
import {MyButton} from "./MyButton";
import {Sidebar} from "./Sidebar";
import {Header} from "./Header";
import {Footer} from "./Footer";
import {Login} from "./Login";

function App() {
    return (
        <div className="App">
            <Login/>
        </div>
    );
}

function test() {
    console.log('test');
}

export default App;
