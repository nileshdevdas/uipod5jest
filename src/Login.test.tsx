import {fireEvent, render, screen, waitFor} from "@testing-library/react";
import App from "./App";
import 'axios';
import React from "react";


describe('Test Login Screen', () => {
    beforeEach(() => {
        global.fetch = require('jest-fetch-mock');
    });

    test('should container username textbox', async () => {
        let app = render(<App/>);
        console.log(fetch.mockResponse(JSON.stringify({"id": 1})));
        await fetch("https://www.google.com").then((result)=>{
            console.log(result.body.toString());
        });
        let usernameElement = await screen.findByTestId('username');
        expect(usernameElement).toBeInTheDocument();
    });

    test('should container password text', async () => {
        let app = render(<App/>);
        let passwordElement = await waitFor(() => screen.findByTestId('password'));
        expect(passwordElement).toBeInTheDocument();
    });

    test('Should Throw Error Message when Submit is clicked without username and password ', async () => {
        let app = render(<App/>);
        let submitButton = await app.findByTestId("login-submit");
        fireEvent.click(submitButton);
        expect(app.queryByText('username error1')).toBeTruthy();
    });
});
