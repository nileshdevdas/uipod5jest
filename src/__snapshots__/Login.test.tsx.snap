// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Login Screen should container username textbox 1`] = `
<div
  className="container"
>
  <form
    className="form"
  >
    <div
      className="form-group"
    >
      <label
        data-testid="username-label"
        htmlFor="username"
      >
        Username
      </label>
      <input
        className="form-control-sm"
        data-testid="username"
        id="username"
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        data-testid="password-label"
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="form-control-sm"
        data-testid="password"
        id="password"
        type="password"
      />
    </div>
    <div
      className="form-group"
    >
      <button
        data-testid="login-submit"
        onClick={[Function]}
        type="button"
      >
        Login
      </button>
    </div>
    <div />
  </form>
</div>
`;
